sync-core/src/main/java/com/cloudant/sync/query/IndexUpdater.java

```diff
@@ -318,6 +318,12 @@ private DBParameter populateDBParameter(List<String> fieldNames,
             Object value = ValueExtractor.extractValueForFieldName(fieldName, rev.getBody());
 
             if (value != null) {
+                // Do not include an empty array field as you would not include a
+                // field with no value.
+                if (value instanceof List && ((List) value).size() == 0) {
+                    continue;
+                }
+
                 includeFieldNames.add(fieldName);
                 args.add(value);
```

Comment:
So, won't an array's value be added to this `args` list twice? It should already be there as part of `initialArgs` on line 309, right? And then  it would get added here again?